sound/soc/sof/intel/hda-ipc.c

```diff
@@ -244,7 +244,8 @@ irqreturn_t hda_dsp_ipc_irq_handler(int irq, void *context)
 	int ret = IRQ_NONE;
```

Comment:
maybe make the commit message more explicit:

ASoC: SOF: Intel: hda: fix MSI issues by merging ipc and stream irq handlers

The existing code uses two handlers for a shared edge-based MSI interrupts. In corner cases,  interrupts are lost, leading to IPC timeouts. Those timeouts do not appear in legacy mode.

This patch merges the two handlers into a single one. Follow-up patches will extend this work to SoundWire interrupts which are also shared with IPC and stream interrupts.